keytree/src/lib.rs

```diff
@@ -94,5 +127,33 @@ mod tests {
 
         assert_eq!(xpub.dk, expected_dk);
         assert_eq!(xpub.point, expected_point);
+        assert_eq!(xpub.precompressed_pubkey, expected_compressed_point);
+    }
+
+    #[test]
+    fn random_xpub_derivation_test() {
+        let seed = [0u8; 32];
+        let mut rng = ChaChaRng::from_seed(seed);
+        let xprv = Xprv::random(&mut rng);
+        let xpub = xprv.to_xpub().to_intermediate_key(customize);
```

Comment:
the usage is going to be mostly via a closure, so maybe something like this is more instructive:

```suggestion
        let xpub = xprv.to_xpub().to_intermediate_key(|t| {
            t.commit_u64("account_id", 42);
        });
```